src/qibocal/plots/allXY.py

```diff
@@ -274,50 +291,55 @@ def msr_beta(folder, routine, qubit, format):
 
         fig.add_trace(
             go.Scatter(
-                x=data.df.beta_param.drop_duplicates().pint.magnitude,  # pylint: disable=E1101
-                y=data.df.groupby("beta_param")["MSR"]  # pylint: disable=E1101
+                x=beta_params,  # pylint: disable=E1101
+                y=data.df.groupby("beta_param", as_index=False)[
+                    "MSR"
+                ]  # pylint: disable=E1101
                 .mean()
                 .pint.to("uV")
                 .pint.magnitude,
-                name=f"q{qubit}/r{i}: avg [Rx(pi/2) - Ry(pi)] - [Ry(pi/2) - Rx(pi)]",
-                marker_color="rgb(100, 0, 255)",
+                name=f"q{qubit}/r{report_n}: Average MSR",
+                marker_color=get_color(report_n),
                 mode="markers",
             ),
             row=1,
             col=1,
         )
 
         # add fitting traces
-        if len(data) > 0 and len(data_fit) > 0:
-            beta_param = np.linspace(
+
+        if len(data) > 0 and (qubit in data_fit.df["qubit"].values):
+            beta_range = np.linspace(
                 min(data.get_values("beta_param", "dimensionless")),
                 max(data.get_values("beta_param", "dimensionless")),
                 20,
             )
-            params = [i for i in list(data_fit.df.keys()) if "popt" not in i]
+
+            params = data_fit.df[data_fit.df["qubit"] == qubit].to_dict(
+                orient="records"
+            )[0]
             fig.add_trace(
                 go.Scatter(
-                    x=beta_param,
+                    x=beta_range,
                     y=cos(
-                        beta_param,
+                        beta_range,
                         data_fit.get_values("popt0"),
                         data_fit.get_values("popt1"),
                         data_fit.get_values("popt2"),
                         data_fit.get_values("popt3"),
                     ),
-                    name=f"Fit q{qubit}/r{i}",
+                    name=f"q{qubit}/r{report_n}: Fit",
                     line=go.scatter.Line(dash="dot"),
                     marker_color="rgb(255, 130, 67)",
```

Comment:
```suggestion
                    marker_color=get_color(4*report_n + 2),
```
To avoid having the same fitting colors when using qq-compare.